#! /bin/sh -e

# DP: Add <builddir>/gcc/ada/rts to the runtime gcc library path.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        cd ${dir} && autoconf2.59
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        cd ${dir} && autoconf2.59
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- configure.in~	2004-11-13 20:06:43.000000000 +0100
+++ configure.in	2004-11-14 09:24:02.000000000 +0100
@@ -2344,9 +2344,9 @@
     *)
       eval "d=\$$RPATH_ENVVAR"
       if test x"$d" != x; then
-	d="$pwd/gcc:$d"
+	d="$pwd/gcc:$pwd/gcc/ada/rts:$d"
       else
-	d="$pwd/gcc"
+	d="$pwd/gcc:$pwd/gcc/ada/rts"
     fi
     SET_GCC_LIB_PATH="\$(RPATH_ENVVAR)=$d; export \$(RPATH_ENVVAR);"
     ;;
